src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -726,28 +726,29 @@ private async Task ConfigureDeploymentFromCli(Recommendation recommendation, Opt
 
             object currentValue = _optionSettingHandler.GetOptionSettingValue(recommendation, setting);
             object? settingValue = null;
-            if (setting.AllowedValues?.Count > 0)
-            {
-                var userInputConfig = new UserInputConfiguration<string>(
-                    idSelector: x => x,
-                    displaySelector: x => setting.ValueMapping.ContainsKey(x) ? setting.ValueMapping[x] : x,
-                    defaultSelector: x => x.Equals(currentValue))
-                {
-                    CreateNew = false
-                };
 
-                var userResponse = _consoleUtilities.AskUserToChooseOrCreateNew(setting.AllowedValues, string.Empty, userInputConfig);
-                settingValue = userResponse.SelectedOption;
-
-                // If they didn't change the value then don't store so we can rely on using the default in the recipe.
-                if (Equals(settingValue, currentValue))
-                    return;
+            if (setting.TypeHint.HasValue && _typeHintCommandFactory.GetCommand(setting.TypeHint.Value) is var typeHintCommand && typeHintCommand != null)

```

Comment:
I switched the order of the `if` statement to give priority to any Type Hint that is available on an option setting as that should preceed everything else.

Reply:
Is there a circumstance that an option would have both an AllowsValues collection and a typehint? Are we introducing that for this change and if so we need to know if there are toolkit ramifications.

Reply:
Originally I didn't have a type hint for the new architecture setting and then I added a typehint for more customization which didn't work, which is when I noticed that the TypeHint should override any default behavior we have on the setting. So for correctness, I have updated the behavior. @awschristou Do you know if there are any issues in the toolkit if we update this behavior? (I don't think there would be any issues currently since we don't have a setting that exercises this user case)

Reply:
I'm a little rusty on this. I believe that TypeHint is a cue for the Toolkit to use custom UIs for a specific property instead of a generic property editor.

I was looking through the Toolkit Code, and AllowedValues is only used in situations where the ValueMapping field does not contain any values.

I think that this should be okay, but I recommend testing out the updated code in the Toolkit to be sure.

Reply:
I tested the changes in the toolkit, though not exhaustively, and there were no issues. Also, once the feature is completed, we will do a bug bash for the toolkit to verify the changes prior to release.